lib/spinor_noise.cu

```diff
@@ -186,12 +162,27 @@ namespace quda {
     }
   }
 
-  void spinorNoise(ColorSpinorField &src, RNG& randstates, QudaNoiseType type)
+  void spinorNoise(ColorSpinorField &src_, RNG& randstates, QudaNoiseType type)
   {
-    switch (src.Precision()) {
-    case QUDA_DOUBLE_PRECISION: spinorNoise<double>(src, randstates, type); break;
-    case QUDA_SINGLE_PRECISION: spinorNoise<float>(src, randstates, type); break;
-    default: errorQuda("Precision %d not implemented", src.Precision());
+    // if src is a CPU field then create GPU field
```

Comment:
does this implicitly transform a CPU field into a GPU field?

Reply:
Not really - it creates a temporary GPU field, runs the hypercubic RNG, copies back into the original CPU field and then deletes the GPU field.